### PR TITLE
Check if result storage implementaion is configurable service

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

File: models/classes/AbstractResultService.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Class AbstractResultService
+ * @package oat\taoResultServer\models\classes\implementation
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+abstract class AbstractResultService extends ConfigurableService implements ResultServerService, ServiceLocatorAwareInterface
+{
+    use OntologyAwareTrait;
+    use ServiceLocatorAwareTrait;
```
Comment (jbout) [inline models/classes/AbstractResultService.php]: ServiceLocatorAwareTrait and ServiceLocatorAwareInterface are already implemented in ConfigurableService

File: models/classes/AbstractResultService.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+/**
+ * Class AbstractResultService
+ * @package oat\taoResultServer\models\classes\implementation
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+abstract class AbstractResultService extends ConfigurableService implements ResultServerService, ServiceLocatorAwareInterface
+{
+    use OntologyAwareTrait;
```
Comment (jbout) [inline models/classes/AbstractResultService.php]: OntologyAwareTrait should not be required

File: models/classes/AbstractResultService.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    use ServiceLocatorAwareTrait;
+
+    /**
+     * @param $class
+     * @return \taoResultServer_models_classes_WritableResultStorage
+     * @throws \common_exception_Error
+     */
+    public function instantiateResultStorage($class)
```
Comment (jbout) [inline models/classes/AbstractResultService.php]: could this function be used outside of OntologyService?

File: models/classes/class.ResultStorageContainer.php [hunk 4]
```diff
@@ -33,14 +37,14 @@ class taoResultServer_models_classes_ResultStorageContainer
...
+     * @param array $implementations
      */
     public function __construct($implementations = array())
     {
         parent::__construct();
+        $resultServerService = ServiceManager::getServiceManager()->get(ResultServerService::SERVICE_ID);
         foreach ($implementations as $key => $implementationParams) {
-            $implementationClassName = $implementationParams["className"];
-            $implementations[$key]["object"] = new $implementationClassName();
+            $implementations[$key]["object"] = $resultServerService->instantiateResultStorage($implementationParams["className"]);
```
Comment (jbout) [inline models/classes/class.ResultStorageContainer.php]: will these implementations be serialized to the session?

File: models/classes/AbstractResultService.php (hunk 3 above)
Comment (hutnikau) [inline models/classes/AbstractResultService.php]: Yes, this function called in \taoResultServer_models_classes_ResultStorageContainer

**Pull request merged**